src/authentication/authentication-hapi-plugin.ts

```diff
@@ -335,6 +412,41 @@ export function findTeamByIdOrName(teamNameOrId: string | number) {
     || team.id === parseInt(_teamNameOrId, 10);
 }
 
+export function authorize(gitlab: GitlabClient) {
+  return async (userName: string, request: Hapi.Request) => {
+    try {
+      let teamId = NaN;
+      // Check if we have a teamId passed in the path
+      teamId = parseInt(request.params.teamId, 10);
+      if (!isNaN(teamId)) {
+        const team = await gitlab.getGroup(teamId, userName);
+        return team.id === teamId;
```

Comment:
I would define a separate `userHasAccessToTeam` function, to make it clear what is happening here.